### PR TITLE
Enhancements for server and client communication

### PR DESCRIPTION
**Pull request by abhiramkulkarni**
Server and client communication is improved to have better security posture.

The calls with no-verify mainly get-server-csr, sign-csr and post-server-cert are removed.

create-server-cert new api is introduced for ease. 

IP dependency is removed and one can generate certificates with hostname, use it during create instance and map it /etc/hosts so that server and client are enabled with mutual tls from start.

### CODE REVIEW

File: client_certificate.py [hunk 1]
```diff
@@ -166,14 +166,78 @@ def client_certificate(self, gen_cert=True):
...
+
     def exit_if_exist(self, *files):
         existing_files = []
         for file in files:
             if os.path.isfile(file):
                 existing_files.append(file)
         if len(existing_files) > 0:
             logging.error('client_certificate: exiting because the following files already exist - {}'.format(' '.join(existing_files)))
-            logging.error('client_certificate: specify CLIENT_CRT_KEY, CAPATH, and CAKEYPATH or empty the directory')
+            logging.error('client_certificate: speficy CLIENT_CRT_KEY, CAPATH, and CAKEYPATH or empty the directory')
```
Comment (shikhagulati14) [inline client_certificate.py:240]: This is not required.

Comment (sreeteja7) [reply client_certificate.py:240]: Yes I see a typo here specify which was original is right than the modified one

**Review by sreeteja7 — APPROVED** [review]
LGTM

**Review by shikhagulati14 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**